src/napari_spatialdata/_widgets.py

```diff
@@ -232,6 +240,65 @@ def _(self, vec: pd.Series, key: str, layer: Layer) -> dict[str, Any]:
         }
 
 
+class ScatterListWidget(AListWidget):
+    attrChanged = Signal()
+    _text = None
+
+    def __init__(self, viewer: Viewer, model: ImageModel, attr: str, **kwargs: Any):
+        AListWidget.__init__(self, viewer, model, attr, **kwargs)
+        self.attrChanged.connect(self._onChange)
+        self.data = -1
+
+    def _onAction(self, items: Iterable[str]) -> None:
```

Comment:
this is the key action of the widget, which instead of creating a new napari layer (with points/labels), it returns the vec for the napari matplotlib, correct? Here you chosed to add this as a method after subclassing AListWidget, to recapitulate the options are:

1. change the `onAction` method	logic of `AListWidget` directly, to make the method behave differently according to some arguemnt (e.g. scatter v. view).
2. add this method directly to `AListWidget` but as a separate method called `_onActionScatter`
3. the way you are doing it now.

Considering that maybe (from first glance) the `ScatterListWidget` could actually be trimmed down to only this method, I think it's worth discussing what's the best option. I think all the 3 are reasonable (yet have different pro and cons). Curious to hear what you end up proposing!